core/src/main/java/io/grpc/StreamTracer.java

```diff
@@ -16,15 +16,15 @@
 
 package io.grpc;
 
-import com.google.errorprone.annotations.DoNotMock;
 import javax.annotation.concurrent.ThreadSafe;
 
 /**
  * Listens to events on a stream to collect metrics.
+ *
+ * <p>DO NOT MOCK: Use TestStreamTracer. Mocks are not thread-safe
```

Comment:
`TestStreamTracer` is an internal interface. Is thread-safety the only reason?  Can I mock it when testing with direct executor?

Reply:
See https://github.com/grpc/grpc-java/issues/4658. I'm purposefully not trying to "change" anything; I'm trying to keep the status-quo. Yes, I see the tester is in internal. There's relatively few users of this, so I'm not too worried about them using something internal (especially inside google where it is easy to track who's using it). But this is also the state the class was already in.

Reply:
If "not to change anything" , "Mocks are not thread-safe" should not be added.
Otherwise LGTM

Reply:
See #3305. The reason for the removal was the thought that mocks aren't thread-safe. Otherwise there is no reasoning why to avoid a mock with this class.